Modules/System/Azure Queues/src/AzureStorageQueuesImpl.Codeunit.al

```diff
@@ -0,0 +1,293 @@
+        MessageText: XmlNode;
+    begin
+        if XmlDocument.ReadFrom(ResponseBody, MessageDocument) then begin
+            if MessageDocument.SelectSingleNode('QueueMessagesList/QueueMessage/InsertionTime', MessageText) then begin
+                exit(MessageText.AsXmlElement().InnerText())
+            end;
+        end;
+    end;
+
+    procedure GetMessageExpirationTimeFromResponseBody(var ResponseBody: Text): Text
+    var
+        MessageDocument: XmlDocument;
+        MessageText: XmlNode;
+    begin
+        if XmlDocument.ReadFrom(ResponseBody, MessageDocument) then begin
+            if MessageDocument.SelectSingleNode('QueueMessagesList/QueueMessage/ExpirationTime', MessageText) then begin
+                exit(MessageText.AsXmlElement().InnerText())
+            end;
+        end;
+    end;
+
+    local procedure GetQueuesListFromResponseBody(var ResponseBody: Text): List of [Text]
+    var
+        MessageDocument: XmlDocument;
+        NamesNodeList: XmlNodeList;
+        Node: XmlNode;
+        NamesList: List of [Text];
+    begin
+        if XmlDocument.ReadFrom(ResponseBody, MessageDocument) then begin
+            if MessageDocument.SelectNodes('EnumerationResults/Queues/Queue/Name', NamesNodeList) then begin
+                if NamesNodeList.Count() > 0 then begin
+                    foreach Node in NamesNodeList do begin
+                        NamesList.Add(node.AsXmlElement().InnerText());
+                    end;
+                    exit(NamesList);
+                end;
+            end;
+        end;
+    end;
+
+
+    procedure DeleteMessageFromQueue(StorageAccountName: Text; Queue: Text[20]; MessageID: Text[100]; PopReceipt: Text[30]): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        HttpClient.Delete(GetAzureQueueUpdateMessageEndpoint(StorageAccountName, Queue, MessageID, PopReceipt), HttpResponse);
+        exit(HttpResponse.IsSuccessStatusCode);
+    end;
+
+    procedure ClearMessagesFromQueue(StorageAccountName: Text; Queue: Text[20]): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        HttpClient.Delete(GetAzureQueueMessagesEndpoint(StorageAccountName, Queue), HttpResponse);
+        exit(HttpResponse.IsSuccessStatusCode);
+    end;
+
+    local procedure GetQueueEndpoint(StorageAccountName: Text; Queue: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
+        exit(EndPoint + '/' + Queue + '/' + AzureQueueSetup."SAS Key");
```

Comment:
The partial string _EndPoint + '/' + Queue + '/'_ occurs several time. Would it be worth considering creating a local procedure for this, e.g. CreateQueueEndpointUrl() where this is returned? This would improve maintainability.